klippy/extras/heater_fan.py

```diff
@@ -37,6 +48,25 @@ def callback(self, eventtime):
             print_time = self.fan.get_mcu().estimated_print_time(curtime)
             self.fan.set_speed(print_time + PIN_MIN_TIME, speed)
         return eventtime + 1.
+    def fan_check(self, eventtime):
+        rpm = self.fan.tachometer.get_status(eventtime)['rpm']
+        has_err = False
+        if rpm is not None:
+            for heater in self.heaters:
+                current_temp, target_temp = heater.get_temp(eventtime)
+                if current_temp > self.heater_temp and rpm < self.min_rpm:
+                    has_err = True
+        if has_err:
+            self.num_err += 1
+            if self.num_err > self.max_err:
+                msg = "'%s' spinning below minimum safe speed of %d rev/min" % (
+                    self.name, self.min_rpm)
+                logging.error(msg)
+                self.printer.invoke_shutdown(msg)
+                return self.printer.get_reactor().NEVER
+        else:
+            self.num_err = 0
+        return eventtime + 1.5
```

Comment:
I have some suggestions to simplify the code

```suggestion
    def fan_check(self, eventtime):
        rpm = self.fan.tachometer.get_status(eventtime)['rpm']
        has_err = False
        if rpm is not None:
            for heater in self.heaters:
                current_temp, target_temp = heater.get_temp(eventtime)
                if current_temp > self.heater_temp and rpm < self.min_rpm:
                    self.num_err += 1
                    break
            else:
                self.num_err = 0
        if self.num_err > self.max_err:
            msg = "'%s' spinning below minimum safe speed of %d rev/min" % (
                self.name, self.min_rpm)
            logging.error(msg)
            self.printer.invoke_shutdown(msg)
            return self.printer.get_reactor().NEVER
        return eventtime + 1.5
```

Reply:
Thank you for this suggestion. I think it also makes the coder easier to read.